hbase-server/src/main/java/org/apache/hadoop/hbase/HDFSBlocksDistribution.java

```diff
@@ -256,6 +256,9 @@ public float getBlockLocalityIndexForSsd(String host) {
   private float getBlockLocalityIndexInternal(String host, Visitor visitor) {
     float localityIndex = 0;
     HostAndWeight hostAndWeight = this.hostAndWeights.get(host);
+    if (hostAndWeight == null) {
+      hostAndWeight = this.hostAndWeights.get(HConstants.LOCALHOST);
```

Comment:
We want to get from the map only when host is localhost? So, we should have condition on `host` rather than having nullable check on `hostAndWeight`.

Reply:
> We want to get from the map only when host is localhost? So, we should have condition on host rather than having nullable check on hostAndWeight.

The logic is, try to get from the map using host first, if failed, then using "localhost". 

> Small UT to test this is possible?

Ok, let me try.